src/components/tz.rs

```diff
@@ -7,33 +7,33 @@ use core::{iter::Peekable, str::Chars};
 
 use num_traits::ToPrimitive;
 
-use crate::components::instant::EpochNanoseconds;
 use crate::{
-    components::{duration::normalized::NormalizedTimeDuration, Instant},
-    iso::{IsoDate, IsoDateTime},
+    components::{duration::normalized::NormalizedTimeDuration, EpochNanoseconds, Instant},
+    iso::{IsoDate, IsoDateTime, IsoTime},
     options::Disambiguation,
-    TemporalError, TemporalResult,
+    TemporalError, TemporalResult, ZonedDateTime,
 };
 
 #[cfg(feature = "experimental")]
 use crate::tzdb::FsTzdbProvider;
 #[cfg(feature = "experimental")]
-use std::sync::{LazyLock, Mutex};
+use std::{
+    ops::Deref,
+    sync::{LazyLock, Mutex},
+};
 
 #[cfg(feature = "experimental")]
 pub static TZ_PROVIDER: LazyLock<Mutex<FsTzdbProvider>> =
     LazyLock::new(|| Mutex::new(FsTzdbProvider::default()));
 
-use super::{instant::is_valid_epoch_nanos, ZonedDateTime};
-
 pub trait TzProvider {
     fn check_identifier(&self, identifier: &str) -> bool;
 
     fn get_named_tz_epoch_nanoseconds(
         &self,
         identifier: &str,
         iso_datetime: IsoDateTime,
-    ) -> TemporalResult<Vec<i128>>;
+    ) -> TemporalResult<Vec<EpochNanoseconds>>;
```

Comment:
I noticed when rebasing this PR that `EpochNanoseconds` can (and probably should) be used in more places throughout `temporal_rs`, then I had initially thought. I added some of that work to this PR due to it somewhat being in the scope of work here. For the rest though, I'm thinking it would be best to open up an issue for it and continue looking into using `EpochNanoseconds` in more places for a subsequent PR.

Reply:
Agreed! Could be a nice first issue for a future contributor.

Reply:
Opened an issue for it 😄